components/infobox/commons/infobox_manufacturer.lua

```diff
@@ -0,0 +1,159 @@
+local Customizable = Widgets.Customizable
+local Builder = Widgets.Builder
+
+---@class ManufacturerInfobox: BasicInfobox
+local Manufacturer = Class.new(BasicInfobox)
+
+---@param frame Frame
+---@return Html
+function Manufacturer.run(frame)
+	local manufacturer = Manufacturer(frame)
+	return manufacturer:createInfobox()
+end
+
+---@return Html
+function Manufacturer:createInfobox()
+	local infobox = self.infobox
+	local args = self.args
+
+	local widgets = {
+		Customizable{
+			id = 'header',
+			children = {
+				Header{
+					name = self:nameDisplay(args),
+					image = args.image,
+					imageDefault = args.default,
+					imageDark = args.imagedark or args.imagedarkmode,
+					imageDefaultDark = args.defaultdark or args.defaultdarkmode,
+					subHeader = self:subHeaderDisplay(args),
+					size = args.imagesize,
+				},
+			}
+		},
+		Customizable{
+			id = 'caption',
+			children = {
+				Center{content = {args.caption}},
+			}
+		},
+		Title{name = (args.informationType or 'Manufacturer') .. ' Information'},
+		Customizable{
+			id = 'formernames',
+			children = {
+				Cell{name = 'Former Name(s)', content = {args.formernames}},
+			}
+		},
+		Cell{name = 'Description', content = {args.description}},
+		Customizable{
+			id = 'seasons',
+			children = {
+				Cell{name = 'Season(s)', content = {args.seasons}},
+			}
+		},
+		Customizable{
+			id = 'enginetotal',
+			children = {
+				Cell{name = 'Engine Total', content = {args.enginetotal}},
+			}
+		},
+		Customizable{
+			id = 'status',
+			children = {
+				Cell{name = 'Status', content = {args.status}},
+			}
+		},
+		Customizable{
+			id = 'history',
+			children = {
+				Builder{
+					builder = function()
+						if args.founded or args.dissolved then
+							return {
+								Title{name = 'History'},
+								Cell{name = 'Founded', content = {args.founded}},
+								Cell{name = 'Dissolved', content = {args.dissolved}}
+							}
+						end
+					end
+				}
+			}
+		},
+		Customizable{id = 'custom', children = {}},
+		Center{content = {args.footnotes}},
+		Customizable{id = 'customcontent', children = {}},
+	}
+
+	infobox:categories('Manufacturers')
+	infobox:categories(unpack(self:getWikiCategories(args)))
+
+	local builtInfobox = infobox:widgetInjector(self:createWidgetInjector()):build(widgets)
+
+	if Namespace.isMain() then
+		self:setLpdbData(args)
+	end
+
+	return builtInfobox
+end
+
+---@param args table
+---@return string[]
+function Manufacturer:getWikiCategories(args)
+	return {}
+end
+
+---@param args table
+---@return string?
+function Manufacturer:nameDisplay(args)
+	return args.name
+end
+
+---@param args table
+function Manufacturer:setLpdbData(args)
+	local name = args.romanized_name or self.name
+
+	local lpdbData = {
+		name = name,
+		region = args.region,
+		logo = args.image,
+		logodark = args.imagedark,
+		locations = Locale.formatLocations(args),
+		extradata = { 
+			status = args.status
+		}
+	}
+		
+	lpdbData = self:addToLpdb(lpdbData, args)
+end
```

Comment:
need to json extradata and the actual storage call is missing too

Reply:
for json would that be 'mw.ext.LiquipediaDB.lpdb_create_json(extradata = {..'

and for the storage call what would be needed?

Reply:
found it:


	lpdbData.extradata = mw.ext.LiquipediaDB.lpdb_create_json(lpdbData.extradata or {})
	mw.ext.LiquipediaDB.lpdb_team('manufacturer_' .. self.name, lpdbData)

Reply:
either use
extradata = mw.ext.LiquipediaDB.lpdb_create_json{key = value, key2 = value2, ...}
or import json module and use
extradata = Json.stringify{key = value, key2 = value2, ...}

Reply:
for the storage call:
mw.ext.LiquipediaDB.lpdb_datapoint(objectName, lpdbData)

Reply:
Is calling the lpdbData.extradata fine or should I just get each value in extradata then?